### PR TITLE
Add some very basic support for type families and instances.

### PR DESCRIPTION
**Pull request by awpr**
This only recognizes that a type family (or data family or associated type family) defines a semantic node, and its name in instances refers to it.  We could have much richer structure analogous to C++ template specialization, but it'd be much more difficult.

Improves the state of affairs of #47.

### CODE REVIEW

File: haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs [hunk 1]
```diff
@@ -292,15 +301,15 @@ declsFromRenamed ctx (hsGroup, _, _, _) =
...
-    hsTyVarBinderName :: HsTyVarBndr id -> IdP id
+    hsTyVarBinderName :: HsTyVarBndr GhcRn -> IdP GhcRn
     hsTyVarBinderName (UserTyVar _ n) = unLoc n
     hsTyVarBinderName (KindedTyVar _ n _) = unLoc n
-    hsTyVarBinderName (XTyVarBndr _) =
-        error "should not hit XTyVarBndr when accessing renamed AST"
+    hsTyVarBinderName (XTyVarBndr no) = noExtCon no
+    -- Data, class, synonym, and family declarations.
+    tyClDecls :: LTyClDecl GhcRn -> [DeclAndAlt]
     -- Datatypes.
```
[the remark this answers is not in the document]
Comment (awpr) [reply haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs]: Done.

**Comment by awpr** [comment]
What's the added benefit of the redundant test coverage with manually-typed-in byte ranges?  It seems like a lot of effort for no purpose when there's already a test that confirms we're generating anchors with precisely the right ranges and graph structure.

**Comment by jinwoo** [comment]
> What's the added benefit of the redundant test coverage with manually-typed-in byte ranges? It seems like a lot of effort for no purpose when there's already a test that confirms we're generating anchors with precisely the right ranges and graph structure.

One benefit I can think of is that unit tests run in CI but the verification tests don't. It was like that at least when I made changes here last time but I'm not sure if it's still true. @robinp Am I correct?

And I'm not sure why this PR doesn't trigger the CI check. @robinp Do you know?

Anyway, I'm fine with only the verification tests if @robinp is okay.

**Comment by robinp** [comment]
CI running: it seems to at https://travis-ci.org/github/google/haskell-indexer/pull_requests, not sure why it doesn't get marked up here. Some webhook to configure no the repository maybe? Could you have a look?

Running unit tests on CI: yes, they run, see 'basic_test' and 'typelink_test' in CI logs output.

Unit vs verification: at one point the idea was that Kythe is just a possible frontend, so the common (translate) layer was unit-tested. Theoretically we could test stuff that Kythe doesn't support there. Also we have some more advanced tests there (FFI?). But on the duplicated tests - well, yes, they are duplicated indeed.

I don't think anything would stop the verification tests to run on CI. If you would wire them up to CI, could skip the dups. Didn't give deep thought.

**Comment by jinwoo** [comment]
Thanks, @robinp . I think this PR is fine to merge then. I don't have a strong opinion about the tests.

And I'll take a look why the CI results are not shown here.

**Pull request merged**